### DISCUSSION TITLE
How to integrate with DNS?

### DISCUSSION THREAD

**Discussion by davehouser1** [netbox-community/netbox#12722] [Help Wanted!]
Hello, I am looking for an IPAM solution that can integrate into DNS. What I mean by this is any IPAM host I add, the hostname will create an A/AAAA + PTR record somewhere. That somewhere could be a local DNS server running on the host (like what PHPIPAM does with pdns), or from afar (manage some MS or bind instance).

Does Netbox support this?

I found this plugin (https://github.com/peteeckel/netbox-plugin-dns), but its not clear if this plugin is just an overlay to keep track of records or if it actually goes out and makes changes to a DNS server. If so how do I integrate?

Hope there is a way to do this. We want to avoid having people RDP into our DNS servers to create records, using NetBox would be great alternative solution!

**Comment by jcollie** [comment]
You'll find that NetBox doesn't have any native tooling to synchronize the data stored in the NetBox database with the "real world". That's deliberate because there's such a variety of systems out there that it would be an impossible task. netbox-plugin-dns just keeps DNS records in the NetBox database. There are places in NetBox for entering DNS names but the native support is very limited.

You'll need to look at 3rd party solutions to synchronize the records stored in NetBox with your DNS server. Since you mentioned RDP I think that it's likely that you are using Microsoft's DNS server. I'm not aware of any existing tools that can synchronize with Microsoft's DNS server. You might start with OctoDNS and see if there's anything available there. Or if you're willing to switch to a different DNS server OctoDNS would be a good place to start.

**Comment by mtinberg** [comment]
I haven't used netbox-plugin-dns but a look through the docs shows some example templates for generating BIND zonefile format from the database using Ansible, so running a local hidden-authoritative ISC BIND DNS service that your Windows DNS service replicates from using native DNS zone replication should be straightforward to setup.

So to answer your question, yes it just stores the relevant information in the database and you have to put together the integration piece yourself.  It looks like there is another plugin which just does dynamic dns updates based on the ip-address dns_name field, if that simplified workflow would be good enough.  Also I think the plugin you found is may be related to https://github.com/auroraresearchlab/netbox-dns


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________

**Reply by jcollie** [reply]
> Also I think the plugin you found is may be related to https://github.com/auroraresearchlab/netbox-dns

Aurora Research Lab apparently has gone out of business or otherwise gone defunct. @peteeckel (who was doing the bulk of the development anyway) forked `netbox-dns` and renamed it `netbox-plugin-dns`. There's zero reason to use the old Aurora Research Lab version.

**Reply by peteeckel** [reply]
Hi @jcollie, thanks for providing that pointer - one little correction, though: There could be a valid reason to stick with `netbox-dns`: The fork that is available as `netbox-plugin-dns` started out with release 0.18.0, which requires NetBox 3.5. If someone is stuck with NetBox 3.4 (or lower) for some reason they will at least temporarily need `netbox-dns`.

If for some reason this situation does not clear up over time I'll provide a 0.17 version as well, but there's not much wrong with `netbox-dns` 0.17.0 for NetBox 3.4 so I think it's safe to use until NetBox can be upgraded to a newer version.

Renaming was unfortunately necessary because Aurora Research Lab disappeared without handing over anything, so the PyPI `netbox-dns` project name is blocked.

**Comment by peteeckel** [comment]
See also https://github.com/peteeckel/netbox-plugin-dns/discussions/19 regarding the idea behind integration between NetBox DNS and actual DNS implementations - much along the lines of what @jcollie already pointed out.

**Comment by davehouser1** [comment]
Thank you all for your suggestions and quick replies. I will continue my discussion over on the netbox-plugin-dns page. 
Being able to engage with an active community such as Netbox is wonderful. It makes all the difference when working with an opensource solution, thanks again!

**Comment by candlerb** [comment]
At home, I use this script I cobbled together to send dynamic DNS updates from Netbox:
https://github.com/candlerb/netbox-webhook-dnsupdate

It only uses the `dns_name` attribute on IPAddress objects, and updates forward + reverse DNS. So if you want to create anything else (e.g. TXT, CNAME, SRV records) then you still do that directly on the DNS server.

**Comment by v0tti** [comment]
I made this small script which syncs NetBox `dns_name` to PowerDNS using the PowerDNS API: https://github.com/v0tti/netbox-powerdns-sync
